src/safeposix/filesystem.rs

```diff
@@ -153,35 +211,31 @@ pub fn load_fs() {
             for serialpair in logvec.drain(..) {
                 let (inodenum, inode) = serialpair;
                 match inode {
-                    Some(inode) => mutmetadata.inodetable.insert(inodenum, inode),
-                    None => mutmetadata.inodetable.remove(&inodenum),
-                };
+                    Some(inode) => {FS_METADATA.inodetable.insert(inodenum, inode);}
+                    None => {FS_METADATA.inodetable.remove(&inodenum);}
+                }
             }
 
             let _logclose = log_fileobj.close();
             let _logremove = interface::removefile(LOGFILENAME.to_string());
 
             // clean up broken links
-            fsck(&mut mutmetadata);
+            fsck();
         }
-
-        // then recreate the log
-        create_log();
-
-
     } else {
-       *mutmetadata = FilesystemMetadata::blank_fs_init();
-       drop(mutmetadata);
-       create_log();
-
-       load_fs_special_files(&utilcage);
-       let metadata = FS_METADATA.read().unwrap();
-       persist_metadata(&metadata);
+        if interface::pathexists(LOGFILENAME.to_string()) {
```

Comment:
Should we let this run if get to this state?

Reply:
I think yes, because it is possible that the user removed one and not the other manually, and I don't think it makes that much sense for us to panic in that case, considering there is only one possible thing they could have wanted to do in this case (run it with a new fs).

Reply:
I don't think we should panic, but maybe cleanup the filesystem and shutdown gracefully with a warning. Then the user can just run again but will explicitly know this is happening.

Also on a similar note, does format_fs get rid of the log if it exists as well?

Reply:
Hmm so right now it warns and keeps going. You'd rather it warn and fail and force them to run it again?
Currently yes it does get rid of the log by calling create_log right after it does the format_fs.

Reply:
Yeah I think that makes more sense.